event_builder.go

```diff
@@ -112,22 +154,18 @@ func (eb *EventBuilder) Build(
 		// marshal them into 'null' instead of '[]', which is bad. Since the
 		// EventBuilder struct is instantiated outside of gomatrixserverlib
 		// let's just make sure that they haven't been left as nil slices.
-		if eventStruct.PrevEvents == nil {
-			eventStruct.PrevEvents = []EventReference{}
-		}
-		if eventStruct.AuthEvents == nil {
-			eventStruct.AuthEvents = []EventReference{}
-		}
+		eventStruct.PrevEvents = toEventReference(eventStruct.PrevEvents)
+		eventStruct.AuthEvents = toEventReference(eventStruct.AuthEvents)
 	case EventFormatV2:
 		// In this event format, prev_events and auth_events are lists of
-		// event IDs as a []string, rather than full-blown []EventReference.
+		// event IDs as a []string, rather than full-blown []eventReference.
 		// Since gomatrixserverlib otherwise deals with EventReferences,
 		// take the event IDs out of these and replace the prev_events and
 		// auth_events with those new arrays.
 		switch prevEvents := eventStruct.PrevEvents.(type) {
 		case []string:
 			eventStruct.PrevEvents = prevEvents
-		case []EventReference:
+		case []eventReference:
```

Comment:
Do we need this? I thought EventFormatV2 == strings only, so this case should never be hit?